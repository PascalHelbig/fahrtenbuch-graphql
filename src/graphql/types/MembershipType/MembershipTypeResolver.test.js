const { makeExecutableSchema } = require('graphql-tools');
const { graphql } = require('graphql');
const MembershipType = require('./MembershipType');
const MembershipTypeResolver = require('./MembershipTypeResolver');

jest.mock('../../../models/Group');
jest.mock('../../../models/User');

const SchemaDefinition = `
  type Query {
    membership: Membership
  }
      
  schema { query: Query }
`;

const testPropterty = (propterty, returnValue) => {
  const mockBookshelf = {
    pivot: {
      get: jest.fn().mockReturnValue(returnValue),
    },
  };
  const resolvers = {
    Query: {
      membership: () => mockBookshelf,
    },
    Membership: MembershipTypeResolver,
  };
  const schema = makeExecutableSchema({
    typeDefs: [SchemaDefinition, MembershipType],
    resolvers,
  });

  const query = `
    query { 
      membership {
        ${propterty}
     }
    }
  `;

  return graphql(schema, query).then((result) => {
    expect(result).toMatchSnapshot();
    return mockBookshelf;
  });
};

it('should get the is_admin from model', () =>
  testPropterty('is_admin', true).then(({ pivot }) =>
    expect(pivot.get).toHaveBeenCalledWith('is_admin')
  )
);

it('should get the user from model', () =>
  testPropterty('user { id }').then(({ pivot }) =>
    expect(pivot.get).toHaveBeenCalledWith('user_id')
  )
);

it('should get the group from model', () =>
  testPropterty('group { id }').then(({ pivot }) =>
    expect(pivot.get).toHaveBeenCalledWith('group_id')
  )
);
