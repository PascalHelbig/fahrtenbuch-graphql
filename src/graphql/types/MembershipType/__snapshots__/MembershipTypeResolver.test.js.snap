exports[`test should get the group from model 1`] = `
Object {
  "data": Object {
    "membership": null,
  },
  "errors": Array [
    [GraphQLError: (intermediate value).fetch is not a function],
  ],
}
`;

exports[`test should get the is_admin from model 1`] = `
Object {
  "data": Object {
    "membership": Object {
      "is_admin": true,
    },
  },
}
`;

exports[`test should get the user from model 1`] = `
Object {
  "data": Object {
    "membership": null,
  },
  "errors": Array [
    [GraphQLError: (intermediate value).fetch is not a function],
  ],
}
`;
